rocky/reports/views.py

```diff
@@ -204,4 +205,19 @@ def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context["report_types"] = self.get_report_types()
         context["report_data"] = self.generate_reports_for_oois()
+        context["report_download_url"] = url_with_querystring(
+            reverse("report_pdf", kwargs={"organization_code": self.organization.code}),
+            True,
+            **self.request.GET,
+        )
         return context
+
+
+class ReportPDFView(ReportView, WeasyTemplateResponseMixin):
+    template_name = "report_pdf.html"
+
+    pdf_filename = "report.pdf"
```

Comment:
Could we do a more descriptive pdf name based on the url arguments used to create it?

Reply:
Yes we can have better filenames... but I am not sure how. A report can include multiple report types and multiple OOIs and we can't add them all to the filename in that case. An option might be to check if there is only one OOI and/or one report type and then add it to the filename. If it's a report with multiple type and multiple OOIs we could do something like use the organisation name and maybe a timestamp.

Maybe we should work on this in a new ticket/PR?